### PR TITLE
allow JSON rpc with array params

### PR DESCRIPTION
**Pull request by wtfti**

### CODE REVIEW

**Review by zetxx — CHANGES_REQUESTED** [review]
you should not make params passed muttable

**Review by kalinkrustev — CHANGES_REQUESTED** [review]
> you should not make params passed muttable

I agree with that. Better put some IF that checks if we have array and copy it. Also there is no point to have {} as non-first parameter in Object.assign

Also you need to assign the request to somebody, who will take care for merging. Otherwise looks you are only asking for review and not for merging.

**Comment by mitushy** [comment]
Updated the PR.

* We are currently completely replacing the ```headers``` parameter if a corsCookie exists and they pass from the frontend unaltered otherwise. Maybe we should consider merging them instead?
* In the case of ```action.params``` being an array, we are attaching a ```headers``` property directly to that array instance.

**Comment by kalinkrustev** [comment]
I am OK with the change, @zetxx please review and merge if you are ok too.

File: react/middleware.js [hunk 1]
```diff
@@ -13,13 +26,11 @@ export default (utBus) => {
             action.methodRequestState = 'requested';
             next(action);
 
-            // Convert action.params to plain js when action.params is immutable, but keeping the original params,
-            // because some reducers require params to stay immutable.
-            var actionParamsJS;
-            if (action.params instanceof immutable.Collection) {
-                actionParamsJS = action.params.toJS();
+            const methodParams = cloneParams(action.params);
```
Comment (zetxx) [inline react/middleware.js]: there is no need to make it constant if you change it later

File: react/middleware.js [hunk 2]
```diff
@@ -13,13 +26,11 @@ export default (utBus) => {
...
+            if (corsCookie) {
+                Object.assign(methodParams, { headers: { 'x-xsrf-token': corsCookie } });
```
Comment (zetxx) [inline react/middleware.js]: you should do:
```js
methodParams = Object.assign(methodParams, { headers: { 'x-xsrf-token': corsCookie } });
```
in order to explicitly show that you overwrite `methodParams`'s value

**Review by zetxx — CHANGES_REQUESTED** [review]

**Pull request merged**